packages/governance/src/binaryBallotCounter.js

```diff
@@ -0,0 +1,121 @@
+// @ts-check
+
+import { assert, details as X } from '@agoric/assert';
+import { makeStore } from '@agoric/store';
+import { makePromiseKit } from '@agoric/promise-kit';
+import { Far } from '@agoric/marshal';
+
+import { E } from '@agoric/eventual-send';
+import { ChoiceMethod, buildBallot } from './ballotBuilder';
+
+const makeWeightedBallot = (ballot, weight) => ({ ballot, weight });
+
+const makeBinaryBallot = (question, positionAName, positionBName) => {
+  const positions = [];
+  assert.typeof(positionAName, 'string');
+  assert.typeof(positionBName, 'string');
+  positions.push(positionAName, positionBName);
+
+  return buildBallot(ChoiceMethod.CHOOSE_N, question, positions, 1);
+};
+
+// Exported for testing purposes
+const makeBinaryBallotCounter = (question, aName, bName) => {
+  const template = makeBinaryBallot(question, aName, bName);
+
+  assert(
+    template.getMethod() === ChoiceMethod.CHOOSE_N,
+    X`Binary ballot counter only works with CHOOSE_N`,
+  );
+  let isOpen = true;
+  const outcomePromise = makePromiseKit();
+  const tallyPromise = makePromiseKit();
+  const allBallots = makeStore('seat');
+
+  const getQuestionPositions = () => ({
+    question,
+    positionA: aName,
+    positionB: bName,
+  });
+
+  const recordBallot = (seat, filledBallot, weight = 1n) => {
+    allBallots.has(seat)
+      ? allBallots.set(seat, makeWeightedBallot(filledBallot, weight))
+      : allBallots.init(seat, makeWeightedBallot(filledBallot, weight));
+  };
+
+  const countVotes = async quorumChecker => {
+    assert(!isOpen, X`can't count votes while the election is open`);
+
+    // ballot template has position choices; Each ballot in allBallots should
+    // match. count the valid ballots and report results.
+    const [positionA, positionB] = template.getPositions();
+    let spoiled = 0n;
+    const tally = {
+      [positionA]: 0n,
+      [positionB]: 0n,
+    };
+
+    allBallots.entries().forEach(([_, { ballot, weight }]) => {
+      const choice = ballot.chosen[0];
+      if (!template.getPositions().includes(choice)) {
+        spoiled += weight;
+      } else {
+        tally[choice] += weight;
+      }
+    });
+
+    const stats = {
```

Comment:
seems to me like the tally should be really actionable. Otherwise how do enforcement mechanisms work?

Reply:
I'm not sure what you mean by "actionable". My expectation was that clients would wait on the outcome. Most things should be able to decide what to do based on that.  If they need the extra data in stats, they can examine it.  What would be more "actionable"?

Reply:
It has the counts. It doesn't have the voting decision.  Where would the code be that implements a required 2/3rds vote for success?

Reply:
> It has the counts. It doesn't have the voting decision.

The outcome is published with a notifier in `outcomePromise`. If you think  `tally` should also include that, I can add that. Would you do that by modifying the existing `stats` object, or constructing a new one to return once the quorum check (line 79) and decision (lines 83-89) have  been made?

> Where would the code be that implements a required 2/3rds vote for success?

The `quorumChecker` (line 79) has access to all the info in the tally.  In the tests, I only have cases where it checks for a strict number, but it's as easy to compare to a fraction.